rewrite-java/src/main/java/org/openrewrite/java/tree/CoordinateBuilder.java

```diff
@@ -246,6 +246,31 @@ public JavaCoordinates replaceMethod() {
         }
     }
 
+    public static class Unary extends Statement {
+        Unary(J.Unary tree) {
+            super(tree);
+        }
+
+        @Override
+        JavaCoordinates after(Space.Location location) {
+            return after(isModifying() ? Space.Location.STATEMENT_PREFIX : Space.Location.EXPRESSION_PREFIX);
+        }
+
+        @Override
+        public JavaCoordinates before() {
+            return before(isModifying() ? Space.Location.STATEMENT_PREFIX : Space.Location.EXPRESSION_PREFIX);
+        }
+
+        @Override
+        public JavaCoordinates replace() {
+            return replace(isModifying() ? Space.Location.STATEMENT_PREFIX : Space.Location.EXPRESSION_PREFIX);
```

Comment:
Not sure if all unary expressions should always be treated as expressions or not by the template compiler. Like method invocations the modifying unary expressions are valid in both contexts. (Also makes me wonder if there may be a similar issue with assignments which can also be used as expressions.)

Reply:
Yeah there is some awkwardness with the tree types which are both statements and expressions. An effort to find bugs with how expression + statements are templated in various contexts would surely succeed at finding quite a variety of bugs.